### PR TITLE
Remove radium from DataBrowser and its dependencies

### PR DESCRIPTION
**Pull request by mikeharv**
8/10 - Now includes: 
* https://github.com/code-dot-org/code-dot-org/pull/47567

This is part of the work to stop using Radium. See https://github.com/code-dot-org/code-dot-org/pull/47367 for more discussion. This PR removes Radium from 19 components which collectively make up the bulk of App Lab's Data Browser:

- DataTable
- EditTableRow
- TableControls
- AddKeyRow
- DataBrowser
- EditTableListRow
- DataOverview
- ConfirmImportButton
- DataTableView
- EditLink
- ConfirmDeleteButton
- LibraryCategory
- EditKeyRow
- ColumnHeader
- DataLibraryPane
- KVPairs
- AddTableListRow
- AddTableRow
- DropdownField

## Links

Google Sheet: [Radium usage in apps/](https://docs.google.com/spreadsheets/d/1chviZU39Qej9_xrc2lrcRrydKJKcs-W6Q0ehzVtH6bg/edit#gid=1128290845)

## Testing story
This work touches all parts of the Data Browser. A comparison video walks through the display and interaction with the components affected. There were no regressions found!
Current Production as of 8/9:

https://user-images.githubusercontent.com/43474485/183724689-1e8e38fc-a1db-4795-9452-c8515820291d.mp4

Localhost as of https://github.com/code-dot-org/code-dot-org/pull/47527/commits/eb1d43b0f0879d459db48033615e2bd06d78a9f5 :

https://user-images.githubusercontent.com/43474485/183724849-1af795c6-1a01-40bb-a8e6-8c6f1645d18a.mp4

There are lots of separate commits here as I want to thoroughly find and test each component. In several cases, Radium could be removed easily. In other cases, scss modules were created in addition to removing Radium. Prior to committing these kinds of changes, I made some visually obvious changes to colors in the new modules to confirm that the module was actually determining the rules used. I looked through the DOM at each affected element to make sure that the expected rules (and if applicable, classes) were present and active.

## Follow-up work

- [x] Several components here import style objects from https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/storage/dataBrowser/dataStyles.js
As follow-up work we would like to convert this .js file into a scss module and then perform another round of updates on these components. ([Slack thread for context](https://codedotorg.slack.com/archives/C03MZ259SUU/p1659988079617719))

Completed! https://github.com/code-dot-org/code-dot-org/pull/47567

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/storage/dataBrowser/column-header.module.scss [hunk 1]
```diff
@@ -0,0 +1,24 @@
+$maxCellWidth: 350px;
+
+.columnName {
+    display: inline-block;
+    max-width: $maxCellWidth;
+    overflow: hidden;
+    white-space: nowrap;
+};
```
Comment (maddiedierker) [inline apps/src/storage/dataBrowser/column-header.module.scss]: a couple of nits: add an empty newline between class definitions (this should be solved for you once we add linting!) and remove the `;`

**Comment by bencodeorg** [comment]
A couple questions off the bat:

1. Were the components where Radium was removed with no other changes (`AddKeyRow`, `AddTableListRow`, `AddTableRow`, maybe more) just not using Radium at all?
2. It might not be worth it at this point since it seems like the changes are pretty straightforward, but would it have been possible to break this up a bit, or do all of these changes need to be made at once?

Maybe if in your list of components you could provide a little categorization of the components and the type of changes you had to make (eg, one category "doesn't actually use Radium", "had to create SCSS module", "just had to use spread operator instead of array syntax", etc)?

File: apps/src/storage/dataBrowser/data-browser.module.scss [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+$borderColor: $lighter_gray;
+$bgColor: $lightest_gray;
+
+.tab {
+    border-color: $borderColor;
+    border-style: solid;
+    box-sizing: border-box;
+    height: $tabHeight;
+    padding: 0 10px;
+    &Active{
```
Comment (maddiedierker) [inline apps/src/storage/dataBrowser/data-browser.module.scss]: nit: add a space before `{`

**Comment by bencodeorg** [comment]
...also, this is amazing. 19 components!!

File: apps/src/storage/dataBrowser/table-controls.module.scss [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    margin-bottom: -28px;
+    // subtract the top margin of the buttonWrapper
+    margin-top: -10px;
+    padding-top: 0px;
+    padding-bottom: 10px;
+    padding-left: 0px;
+    padding-right: 0px;
+    // make the buttons align right usually, but align left if they
+    // are forced to wrap onto the next line by a very long table name.
+    text-align: justify
```
Comment (maddiedierker) [inline apps/src/storage/dataBrowser/table-controls.module.scss:27]: need a `;` at the end of this rule

File: apps/src/storage/dataBrowser/data-table.module.scss [hunk 4]
```diff
@@ -0,0 +1,21 @@
+@import 'color.scss';
+
+$minTableWidth: 600px;
+
+.addColumnHeader {
+    width: 19
```
Comment (maddiedierker) [inline apps/src/storage/dataBrowser/data-table.module.scss]: need a `;` at the end of this rule

**Review by maddiedierker — COMMENTED** [review]
there are a few rules that need `;` at the end to make sure we don't break subsequent rules, so make sure to do a pass for those! adding linting will help with this so we don't have to do it manually 😄

**Comment by mikeharv** [comment]
@bencodeorg The components where Radium was simply removed were not using it.

I went back and forth on the correct number of pull requests for these changes. I think it may have been an option to put out each separately, but it felt like it would be inefficient in terms of workflow. Some of these components depend on each other and/or interact closely on the website so it seemed simplest to test things together. I was also weary of having to write up 19 repetitive PRs, and requesting a lot of separate reviews. If any of these changes cause regressions with the Data Browser that I missed, I think I'd rather quickly revert the whole thing than have to immediately dig up exactly which component is causing the problem.

Happy to do things differently next time!

File: apps/src/storage/dataBrowser/column-header.module.scss (hunk 1 above)
Comment (mikeharv) [reply apps/src/storage/dataBrowser/column-header.module.scss]: Thanks! I'll make a pass on all 8 modules for this.

File: apps/src/storage/dataBrowser/table-controls.module.scss (hunk 3 above)
Comment (mikeharv) [reply apps/src/storage/dataBrowser/table-controls.module.scss:27]: I am seeing a few others as well. I'll do a thorough pass now.

**Comment by bencodeorg** [comment]
> @bencodeorg The components where Radium was simply removed were not using it.
> 
> I went back and forth on the correct number of pull requests for these changes. I think it may have been an option to put out each separately, but it felt like it would be inefficient in terms of workflow. Some of these components depend on each other and/or interact closely on the website so it seemed simplest to test things together. I was also weary of having to write up 19 repetitive PRs, and requesting a lot of separate reviews. If any of these changes cause regressions with the Data Browser that I missed, I think I'd rather quickly revert the whole thing than have to immediately dig up exactly which component is causing the problem.
> 
> Happy to do things differently next time!

Oh definitely not suggesting 19 PRs, that would be a lot. I think just a bit of a guide about how a reviewer should think about looking at changes when there are this many files changed (eg, a categorization of the types of changes made) can be helpful.

**Comment by mikeharv** [comment]
@madelynkasula Please find your requested changes in this newest commit: https://github.com/code-dot-org/code-dot-org/pull/47527/commits/42fe57b0d3d634281079fff8864bb345a902b18d

File: apps/src/storage/dataBrowser/ColumnHeader.jsx [hunk 5]
```diff
@@ -115,33 +116,31 @@ class ColumnHeader extends React.Component {
...
-    ];
+    };
     return (
       <th style={dataStyles.headerCell} className="uitest-data-table-column">
-        <div style={containerStyle} className="flex">
-          <div style={styles.columnName} className="test-tableNameDiv">
+        <div
+          className={classNames(
+            style.container,
+            this.props.isEditing ? style.containerIsEditing : ''
```
Comment (bencodeorg) [inline apps/src/storage/dataBrowser/ColumnHeader.jsx]: Nit: the way I've been seeing this done is `this.props.isEditing && styles.containerIsEditing`

File: apps/src/storage/dataBrowser/column-header.module.scss [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+.container {
+    justify-content: space-between;
+    padding: 6px 0px;
+    display: flex;
+    &IsEditing {
+        display: none;
+    }
+}
+
+iconWrapper {
```
Comment (bencodeorg) [inline apps/src/storage/dataBrowser/column-header.module.scss]: Missing period?

**Review by bencodeorg — COMMENTED** [review]
Looks good to me! A couple nits but this is crazy progress 🙌

File: apps/src/storage/dataBrowser/ColumnHeader.jsx (hunk 5 above)
Comment (mikeharv) [reply apps/src/storage/dataBrowser/ColumnHeader.jsx]: Changed!

File: apps/src/storage/dataBrowser/column-header.module.scss (hunk 6 above)
Comment (mikeharv) [reply apps/src/storage/dataBrowser/column-header.module.scss]: Great catch! Fixed!

**Review by bencodeorg — APPROVED** [review]

**Review by maddiedierker — APPROVED** [review]
🎉 great work, mike!

**Pull request merged**